pkg/bpf/map.go

```diff
@@ -747,21 +745,14 @@ func ConvertKeyValue(bKey []byte, bValue []byte, key interface{}, value interfac
 // MetadataDiff compares the metadata of the BPF maps and returns false if the
 // metadata does not match
 func (m *Map) MetadataDiff(other *Map) bool {
-	if m == nil || other == nil {
+	switch {
+	case m == other:
+		return true
+	case m == nil || other == nil:
 		return false
+	default:
+		return m.DeepEquals(other)
```

Comment:
This is changing the behavior of the function; previously, it ignored the file descriptor, but DeepEquals on Map checks whether the MapInfo is equal in both (which includes the fd).

Reply:
That's not true, the previous code did change `Map.fd`, which was ignored in `DeepEquals`. The setting of `Map.fd` to 0 was only to simplify the diff.

Reply:
You're right, and I'm wrong :)